fix/categories

AlluringUnicorn opened the pull request:
Пофіксила баг з кольором, баг з тим що картка закривала дроп даун, додала свгшку, додала підтягування категорій з серверу в кнопки і дроп даун ліст

AlluringUnicorn committed:
fix/categories

snizhanamohyl commented on src/partials/categories.html:
```diff
@@ -1,46 +1,51 @@
-      <li><button class="dropdown-item" type="button">Lens</button></li>
+      <li><button class="dropdown-item" type="button"></button></li>
+      <li><button class="dropdown-item" type="button"></button></li>
+      <li><button class="dropdown-item" type="button"></button></li>
```
порожніх лішок в ідеалі бути не повинно, вони мають додаватися через js
і там наче прийшов масив з 50 елементів, відображаються не всі категорії, чи то скрол не працює

AlluringUnicorn replied:
Порожні лішки в дропдаун прибрала, а в кнопках це не можливо через кнопку Others яка має бути там першочергово, якщо лишити її там першочергово, а через js рендерити всі інші, то по перше контент при завантаженні дуже скаче, а по друге, інші лішки рендеяться в неправильному порядку, і також деякі лішки мають різні класи для їх додавання/ прибирання на різних ширинах екрану
якщо ж її також рендерити через js, то на фазі парсу коду він не бачить eventlistener на цій кнопці others якої на той момент ще не існує і відмовляється взагалі виконувати " не валідний" код

snizhanamohyl reviewed (changes requested):
файлик categories.js має бути у папці js

AlluringUnicorn committed:
fix/categories/bugs, add features

Перемістила js файл в папку js.
Додала скролл всіх категорій
Лішки дропдаун лісту ренедеряться через js
Додала курсор поінтер свг з макету
Прибрала баг реагування курсору на закритий дропдаун ліст
Додала ренедер всіх з початку категорій на мобільній версії в кнопку Categories замість Others як на інших версіях
Пофіксила падінги дродаун лісту на моб версії

AlluringUnicorn committed:
Merge branch 'main' into fix/categories

snizhanamohyl committed:
Merge branch 'main' into fix/categories
